multinet/db.py

```diff
@@ -154,20 +155,55 @@ def workspace_table(workspace: str, table: str, offset: int, limit: int) -> dict
 
     query = f"""
     FOR d in {table}
-      LIMIT {offset}, {limit}
-      RETURN d
+        LIMIT {offset}, {limit}
+        RETURN d
+    """
+
+    count = workspace_table_row_count(workspace, table)
+    rows = aql_query(workspace, query)
+
+    return {"count": count, "rows": list(rows)}
+
+
+def workspace_table_rows(
+    workspace: str, table: str, offset: int, limit: int
+) -> Generator[Dict, None, None]:
+    """Stream the rows of a table in CSV form."""
+
+    query = f"""
+    FOR d in {table}
+        LIMIT {offset}, {limit}
+        RETURN d
     """
 
+    return aql_query(workspace, query)
+
+
+def workspace_table_row_count(workspace: str, table: str) -> int:
+    """Return the number of rows in a table."""
     count_query = f"""
     FOR d in {table}
         COLLECT WITH COUNT INTO count
         return count
     """
+    return next(aql_query(workspace, count_query))
 
-    count = aql_query(workspace, count_query)
-    rows = aql_query(workspace, query)
 
-    return {"count": list(count)[0], "rows": list(rows)}
+def workspace_table_keys(
+    workspace: str, table: str, filter_keys: bool = False
+) -> List[str]:
+    """Get the keys of a table in a workspace."""
+
+    query = f"""
+    FOR d in {table}
+        LIMIT 0, 1
+        RETURN ATTRIBUTES(d)
+    """
+    cursor = next(aql_query(workspace, query))
```

Comment:
Even though it's not supposed to happen, we need to guard against the case when there are no rows in the table. We should actually test what happens when we upload a CSV file that has only a header row, and no data rows.